File: frontend/src/preview/preview.ts
import {actions} from './preview-actions'
import {render} from './preview-ui'
import {PreviewState} from "../state";
import {ScenarioRow} from "../models";

import {patch} from '../common/snabbdom'

export function Preview(element, scenarioRow: ScenarioRow) {
  let node = element

  function renderState(state: PreviewState) {
    node = patch(node, render(state, actionHandler(state)))
  }

  function actionHandler(state: PreviewState) {
    return function (action) {
      const promise = actions(state, action)
      if (promise && promise.then)
        promise.then(renderState)
    }
  }

  renderState({
    scenarioRow: scenarioRow
  })

  return this
}

File: frontend/src/main/scala/roborace/frontend/preview/Preview.scala
//package roborace.frontend.preview
//
//import roborace.frontend.PreviewState
//import roborace.frontend.util.SnabbdomApp
//import gameEntities.ScenarioResponse
//import org.scalajs.dom.raw.Element
//
//class Preview(container: Element, scenarioResponse: ScenarioResponse) extends SnabbdomApp{
//  def renderState(state: PreviewState): Unit = {
//    patch(container, PreviewUi.render(state))
//  }
//
//  renderState(PreviewState(scenarioResponse))
//}
